package.json

```diff
@@ -52,9 +52,7 @@
   },
   "dependencies": {
     "arrify": "^2.0.0",
-    "extend": "^3.0.1",
-    "split-array-stream": "^2.0.0",
-    "stream-events": "^1.0.4"
```

Comment:
The reason we use `stream-events` is to wait for when data is actually asked of the stream until we begin the authentication process. Removing that could be breaking-change-ish, since the user would now be required to immediately invoke the stream to guarantee valid credentials (unless retry logic and invalid credential error handling is now automatic in this scenario)

Reply:
I don't think that has changed technically. Since we're implementing a `_read` method doesn't that mean we only begin making requests once the user attempts to read from the stream?

Reply:
Yep, you're right. My head must be stuck in the way we used to do it, where we had a `Connection` class. Or, I'm just losing my mind.

Reply:
Before I think we used a `through` stream, which yeah `stream-events` would be the best way to tap into that event.